### DISCUSSION TITLE
Problem Running Codemods to update the project

### DISCUSSION THREAD

**Discussion by hanan-commits-work** [iTwin/itwinjs-core#2956] [Q&A] [answered]
I am upgrading the project from 2.19.x to 3.0, and I am having trouble upgrading the project using codemods.

Here is the command to run codemods.
../codemods/run-codemods.bat PACKAGE.JSON_PATH TSCONFIG_PATH SRC_PATH

I added the above script in package.json using the relative path. But I keep running into error, "_codemons-main is not recognized as an internal or external command_".

I then tried to run it using cmd, I keep getting "_the system cannot find the path specified_".

Can anyone tell me what I am doing wrong or have any solutions on how to fix this problem?

Thank You.
Abdul Hanan

**Comment by pmconne** [comment]
Did you install as described in [step 2 of the README](https://github.com/iTwin/codemods#readme)?

> npm install --production

**Reply by hanan-commits-work** [reply]
Yes it is installed.

**Comment by pmconne** [comment, marked as answer]
Did you try running it from the command line as described in the README? The following makes it sound like you're trying to run it as an npm script from some package.json:

> I added the above script in package.json using the relative path

**Reply by hanan-commits-work** [reply]
Yes, I was running from cmd, but due to errors, I tried running on npm script.

I keep getting "_codemons-main is not recognized as an internal or external command_ ,or _..is not recognized as an internal or external command_ " using command line.

**Reply by pmconne** [reply]
> codemons-main is not recognized as an internal or external command

I can find no occurrences of the string "codemons-main" in the repository after installation. Can you show your actual command line please?

**Reply by hanan-commits-work** [reply]
codemods-main is the folder name of codemods after I cloned it from the repo,
Here is the command line
![Screenshot 2021-12-28 224117](https://user-images.githubusercontent.com/89520756/147592661-fe0475a2-8209-496e-86ab-65a3d372fc47.png)

**Reply by bbastings** [reply]
Wrong slash? Should be: codemods-main\run-codemods...

**Reply by aruniverse** [reply]
Where did you clone the codemods repo to? Take a look at [step 1 in the readme](https://github.com/iTwin/codemods#readme).

Your directory structure should look like:
|───SomeTopLevelDirectory
|   ├───codemods
|   └───web-viewer-3

then from within the web-viewer-3 dir you can run `../codemods/run-codemods.bat PACKAGE.JSON_PATH TSCONFIG_PATH SRC_PATH`

I'm guessing you cloned the codemod repo as codemods-main on your end, is that right?

**Reply by hanan-commits-work** [reply]
Yes thats right
Let me try it

**Reply by hanan-commits-work** [reply]
So the previous error is resolved thank you all.

But now Im getting path not found error. But package.json is getting updated. Kindly look at this.
![Screenshot 2021-12-29 103651](https://user-images.githubusercontent.com/89520756/147630582-f8d978c3-be2b-4c11-82ba-7694a4ba7c5e.png)

**Reply by hanan-commits-work** [reply]
It seems that system is not finding the paths to tsconfig.json and src folder

**Comment by hanan-commits-work** [comment]
Thank You.
Your solutions worked.
